monitoring/resourse_lakehouse_monitor.go

```diff
@@ -0,0 +1,139 @@
+package monitoring
+
+import (
+	"context"
+
+	"github.com/databricks/databricks-sdk-go/service/catalog"
+	"github.com/databricks/terraform-provider-databricks/common"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func removeSnapshotField(originalSchema map[string]*schema.Schema) map[string]*schema.Schema {
+	tmpSchema := make(map[string]*schema.Schema)
+	for k, v := range originalSchema {
+		tmpSchema[k] = v
+	}
+	delete(tmpSchema, "snapshot")
+	return tmpSchema
+}
+
+type Snapshot struct {
+}
+type Monitor struct {
+	AssetsDir                string                                   `json:"assets_dir"`
+	BaselineTableName        string                                   `json:"baseline_table_name,omitempty"`
+	CustomMetrics            []catalog.MonitorCustomMetric            `json:"custom_metrics,omitempty"`
+	DataClassificationConfig *catalog.MonitorDataClassificationConfig `json:"data_classification_config,omitempty"`
+	FullName                 string                                   `json:"table_name"`
+	InferenceLog             *catalog.MonitorInferenceLogProfileType  `json:"inference_log,omitempty"`
+	Notifications            []catalog.MonitorNotificationsConfig     `json:"notifications,omitempty"`
+	OutputSchemaName         string                                   `json:"output_schema_name"`
+	Schedule                 *catalog.MonitorCronSchedule             `json:"schedule,omitempty"`
+	Snapshot                 Snapshot                                 `json:"snapshot,omitempty"`
```

Comment:
If `Snapshot` is optional, then it should be a pointer, not struct: 

```
Snapshot                 *Snapshot                                 `json:"snapshot,omitempty"`
```

Reply:
This is a bug in the Go SDK. The Go SDK currently treats empty structures and components that can take on any value in the same way. I am opening a PR to fix this here: https://github.com/databricks/databricks-sdk-go/pull/821